src/fiber/gossip.rs

```diff
@@ -1017,83 +1018,30 @@ impl<S: GossipMessageStore> ExtendedGossipMessageStoreState<S> {
     // We will also change the relevant state (e.g. update the latest cursor).
     // The returned list may be sent to the subscribers.
     async fn prune_messages_to_be_saved(&mut self) -> Vec<BroadcastMessageWithTimestamp> {
-        let complete_messages = self
-            .messages_to_be_saved
-            .iter()
-            .filter(|m| self.has_dependencies_available(m))
-            .cloned()
-            .collect::<HashSet<_>>();
-        self.messages_to_be_saved
-            .retain(|v| !complete_messages.contains(v));
-
-        let mut sorted_messages = Vec::with_capacity(complete_messages.len());
-
-        // Save all the messages to a map so that we can easily order messages by their dependencies.
-        let mut messages_map: HashMap<
-            (BroadcastMessageID, bool),
-            VecDeque<BroadcastMessageWithTimestamp>,
-        > = HashMap::new();
-
-        for new_message in complete_messages {
-            let key = (
-                new_message.message_id(),
-                match &new_message {
-                    // Message id alone is not enough to differentiate channel updates.
-                    // We need a flag to indicate if the message is an update of node 1.
-                    BroadcastMessageWithTimestamp::ChannelUpdate(channel_update) => {
-                        channel_update.is_update_of_node_1()
-                    }
-                    _ => true,
-                },
-            );
-            let messages = messages_map.entry(key).or_default();
-            let index = messages.partition_point(|m| m.cursor() < new_message.cursor());
-            match messages.get(index + 1) {
-                Some(message) if message == &new_message => {
-                    // The same message is already saved.
-                    continue;
-                }
-                _ => {
-                    messages.insert(index, new_message);
-                }
-            }
-        }
+        let messages_to_be_saved = take(&mut self.messages_to_be_saved);
+        let (complete_messages, uncomplete_messages) = messages_to_be_saved
+            .into_iter()
+            .partition(|m| self.has_dependencies_available(m));
```

Comment:
There is a subtle bug here. Because we replaces `messages_to_be_saved` in line 1021, the line `1024` will always return false for `ChannelUpdate` messages that do not have corresponding `ChannelAnnouncement` in the store. Previously, the behavior is to also check corresponding `ChannelAnnouncement` in `messages_to_be_saved`. This bug is fixed in https://github.com/contrun/fiber/commit/8ca34e017989fb0c7f513751bff250faee2ae777 . I will create a PR later.